File: cheque-management-app/src/main/java/com/darmi/chequemanagementapp/service/ChequeService.java
package com.darmi.chequemanagementapp.service;

import com.darmi.chequemanagementapp.entity.ChequeEntity;
import com.darmi.chequemanagementapp.model.Cheque;
import com.darmi.chequemanagementapp.model.Customer;
import org.springframework.transaction.annotation.Transactional;

import java.text.ParseException;
import java.time.LocalDate;
import java.util.List;

public interface ChequeService {
    Cheque addCheque(Cheque cheque) throws ParseException;

    List<Cheque> getAllCheques();

    @Transactional
    Boolean deleteCheque(long id);

    Cheque editCheque(long id, Cheque cheque);

    Cheque getChequeById(long id);

    List<Cheque> search(String keyword);

    List<Cheque> filterByStatus(ChequeEntity.Status status);

    double getTotalAmountOfPendingCheques();

    List<Cheque> getWeekCheques();

    double getTotalAmountByClearingDateRange(LocalDate startDate, LocalDate endDate);

    List<Cheque> filterByClearingDate(LocalDate startDate, LocalDate endDate);
}

File: cheque-management-system-ui/src/service/chequeService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/cheques";

class chequeService {
  saveCheque(cheque) {
    return axios.post(BASE_URL, cheque);
  }
  getAllCheques() {
    return axios.get(BASE_URL);
  }
  deleteCheque(id) {
    return axios.delete(BASE_URL + "/" + id);
  }
  editCheque(id, cheque) {
    return axios.put(BASE_URL + "/" + id, cheque);
  }
  getChequeById(id) {
    return axios.get(BASE_URL + "/" + id);
  }
  searchByKeyword(keyword) {
    return axios.get(BASE_URL + "/keyword/" + keyword);
  }
  filterByStatus(status) {
    return axios.get(BASE_URL + "/status/" + status);
  }
  getTotalAmountOfPendingCheques() {
    return axios.get(BASE_URL + "/totalPendingAmount");
  }
  getWeekCheques() {
    return axios.get(BASE_URL + "/weekCheques");
  }
  filterByClearingDate(startDate, endDate) {
    return axios.get(BASE_URL + "/" + startDate + "/" + endDate);
  }
  getTotalAmountOfPendingByClearingDateRange(startDate, endDate) {
    return axios.get(BASE_URL + "/totalAmount/" + startDate + "/" + endDate);
  }
}

// eslint-disable-next-line import/no-anonymous-default-export
export default new chequeService();
